src/viam/app/location/client.py

```diff
@@ -0,0 +1,720 @@
+        return [LocationClient.RobotPart.robot_part_from_proto(robot_part=part) for part in response.parts]
+
+    async def get_robot_part(self, robot_part_id: Optional[str], dest: Optional[str] = None, indent: Optional[int] = 4) -> RobotPart:
+        """Get a robot part.
+
+        Args:
+            robot_part_id (str): ID of the robot part to get.
+            dest (Optional[str]): Optional filepath to write the robot part's config file in JSON format to.
+            indent (Optional[int]): Optional size (in number of spaces) of indent when writing config to `dest`. Defaults to 4.
+
+        Raises:
+            GRPCError: If an invalid robot part ID is passed.
+
+        Returns:
+            LocationClient.RobotPart: The robot part.
+        """
+        request = GetRobotPartRequest(id=robot_part_id)
+        response: GetRobotPartResponse = await self._location_client.GetRobotPart(request, metadata=self._metadata)
+
+        if dest:
+            try:
+                file = open(dest, "w")
+                file.write(f"{json.dumps(json.loads(response.config_json), indent=indent)}")
+            except Exception as e:
+                LOGGER.err(f"Failed to write config JSON to file {dest}", exc_info=e)
+
+        return LocationClient.RobotPart.robot_part_from_proto(robot_part=response.part)
+
+    # TODO: Figure out what a page_token is.
+    async def get_robot_part_logs(
+        self,
+        robot_part_id: str,
+        filter: Optional[str] = None,
+        dest: Optional[str] = None,
+        errors_only: Optional[bool] = False,
+    ) -> List[LogEntry]:
+        """Get the logs associated with a robot part.
+
+        Args:
+            robot_part_id (str): ID of the robot part to get logs from.
+            filter (Optional[str]): Only include logs with messages that contain the string `filter`. Defaults to empty string "" (i.e., no
+                filter).
+            dest (Optional[str]): Optional filepath to write the log entries to.
+            errors_only (Optional[bool]): Optional boolean specifying whether or not to only include error logs. Defaults to False.
+
+        Raises:
+            GRPCError: If an invalid robot part ID is passed.
+
+        Returns:
+            List[LocationClient.LogEntry]: The list of log entries.
+        """
+        page_token = ""
+        logs = []
+
+        while True:
+            request = GetRobotPartLogsRequest(id=robot_part_id, errors_only=errors_only, filter=filter, page_token=page_token)
+            response: GetRobotPartLogsResponse = await self._location_client.GetRobotPartLogs(request, metadata=self._metadata)
+            if not response.next_page_token or len(response.next_page_token) == 0:
+                break
+            else:
+                logs += [LocationClient.LogEntry.log_entry_from_proto(log_entry=log_entry) for log_entry in response.logs]
```

Comment:
If you're getting ALL the logs ever on a robot, this could be a huge list. Do we want to default to returning the entire list? Or should we use the pagination that's already built into the request?